csharp/extractor/Semmle.Extraction.CSharp.DependencyFetching/DependencyManager.Nuget.cs

```diff
@@ -148,7 +182,20 @@ private void RestoreProjects(IEnumerable<string> projects, out IEnumerable<strin
             CompilationInfos.Add(("Failed project restore with package source error", nugetSourceFailures.ToString()));
         }
 
-        private void DownloadMissingPackages(List<FileInfo> allFiles, ISet<string> dllPaths, bool withNugetConfig = true)
+        private void DownloadMissingPackagesFromSpecificFeeds(List<FileInfo> allNonBinaryFiles, HashSet<string> dllPaths)
+        {
+            var reachableFallbackFeeds = GetReachableFallbackNugetFeeds();
+            if (reachableFallbackFeeds.Count > 0)
+            {
+                DownloadMissingPackages(allNonBinaryFiles, dllPaths, withNugetConfigFromSrc: false, fallbackNugetFeeds: reachableFallbackFeeds);
+            }
+            else
+            {
+                logger.LogWarning("Skipping download of missing packages from specific feeds as no fallback Nuget feeds are reachable.");
+            }
+        }
+
+        private void DownloadMissingPackages(List<FileInfo> allFiles, ISet<string> dllPaths, bool withNugetConfigFromSrc = true, IEnumerable<string>? fallbackNugetFeeds = null)
```

Comment:
Is the `withNugetConfigFromSrc` needed? Isn't the case that if a list of `fallbackNugetFeeds` is provided when we use them.